wasi/wasi.go

```diff
@@ -170,6 +173,30 @@ func Args(args []string) (Option, error) {
 	}, nil
 }
 
+// Environ returns an option to set environmental variables to the WASIEnvironment.
+// * keys: the names of environment variables to set
+// * values: the corresponding values of environment variables to set
+// Environ returns an error if the length of the keys and values doesn't match,
+// or the length or the total size of those strings exceed the max of uint32.
+func Environ(keys []string, values []string) (Option, error) {
```

Comment:
is it valid to have duplicated keys?

Reply:
This make me wonder whether it would be better to have `map[key]value` as inputs as I imagine users usually hold environments as maps rather than index aligned two slices like this

Reply:
oh maybe this is dup from @codefromthecrypt's review :D

Reply:
I also thought about dupes, but I doubt the spec will actually enforce this. Also, pros and cons on this are easier if we defer to os.Environ (ex this was go's fault not ours ;P)

Reply:
also if we do map insertion order is lost which may be fine, but also complicates. dupe could be checked internally if we wanted to

Reply:
I replied about the part of the discussion on `map[key]value` vs `os.Environ` in another thread.

Reply:
As for the question whether a duplicated environment variable is allowed or not, I doubt WASI has any rule on duplication right now. So, there may not be any guarantee from the perspective of WASI spec that the result of args_get is unique, though it will be also ok for wazero to accept only unique environment variables, mentioning that in the rationale note. Anyway, we don't have to think about this behavior by accepting the same format as `os.Environ`.

side note: incidentally, operating systems such as Linux don't guarantee the uniqueness of the environment variables, though WASI may not follow that weird spec.

Reply:
PS if anyone ever finds any value validation of anything in wasi, please point it out. So far, you are lucky if a name is defined for a parameter.